staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go

```diff
@@ -130,8 +131,23 @@ func ValidateNoNewFinalizers(newFinalizers []string, oldFinalizers []string, fld
 // ValidateImmutableField validates the new value and the old value are deeply equal.
 func ValidateImmutableField(newVal, oldVal interface{}, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
+	defer func() {
+		if r := recover(); r != nil {
+			// If we ever get here, then either DeepEqual or cmp.Diff failed.
+			// The former is unlikely and indicates a serious problem with the data,
+			// the latter only gets called when we already know that there is a difference.
+			// In both cases we reject the new value.
+			allErrs = append(allErrs, field.Invalid(fldPath, newVal, FieldImmutableErrorMsg))
```

Comment:
All the errors above were `field.Forbidden` before

Reply:
Do you mean that I should write `allErrs = append(allErrs, field.Forbidden(fldPath, FieldImmutableErrorMsg))`?